### PR TITLE
Bump everything

### PR DESCRIPTION
**Pull request by sj26**
This gem is holding back our upgrade to Faraday 0.13.1 because of the gemspec's version requirement, but it seems to work fine. I've bumped the requirements and made sure backwards compat is preserved using bundler for the travis tests but new rubies will use the latest faraday et al.

I also tidied up some rubocop warnings and fixed a failing test which stubs ENV.

🙇

### CODE REVIEW

**Comment by sj26** [comment]
@sferik hey there, sorry to ping you directly but it looks like you may have been doing a bit of maintenance on this repo. Would you consider this PR? There's no functional change, it just improves compatibility with some new gem versions.

**Comment by AlfonsoUceda** [comment]
is there a chance to get this PR merged? Thanks!

**Comment by pboling** [comment]
@sj26 I will merge this ASAP once the conflicts are resolved!  I am the new maintainer of this project.

**Comment by sj26** [comment]
🙇🏻👏

Do you need me to address the conflicts?

**Comment by pboling** [comment]
@sj26 If you have the time, that would be awesome.  Just rebase on latest master.  Otherwise I'll get to it when I can.

**Comment by sj26** [comment]
@pboling I've rebased this on latest master, which means this is just spec cleanups.

**Comment by pboling** [comment]
If anyone has time to fix the three failing rubo cop errors I will merge this.

**Comment by pboling** [comment]
@sj26 thanks!  I will prepare for a new release tonight.

File: Gemfile [hunk 1]
```diff
@@ -13,21 +26,23 @@ group :test do
...
-  platforms :ruby_18, :ruby_19 do
+  if ruby_engine == 'ruby' && ruby_version < Gem::Version.new('2.0')
     gem 'json', '< 2.0'
     gem 'term-ansicolor', '< 1.4.0'
     gem 'tins', '< 1.7'
   end
+  gem 'rack', '~> 1.2' if ruby_version < Gem::Version.new('2.2')
+
+  # Rubocop only works on new ruby
+  gem 'rubocop', '>= 0.37' if ruby_version >= Gem::Version.new('2.1')
```
Comment (bquorning) [inline Gemfile]: I’m not sure why RuboCop was reverted from the locked down version `~> 0.50.0` to this much larger range of versions? This is the opposite of what was fixed in #322.

Comment (pboling) [reply Gemfile]: Great point.  I will see what happens when I set it back to `'~> 0.50.0'`

Comment (bquorning) [reply Gemfile]: I’m working on a PR :-)

Comment (sj26) [reply Gemfile]: Bad rebase, I think. Sorry!

**Pull request merged**